packages/connect-form/src/components/advanced-options-tabs/authentication-tab/authentication-tab.tsx

```diff
@@ -69,27 +81,65 @@ const contentStyles = css({
   width: '50%',
 });
 
+function getSelectedAuthTabForConnectionString(
+  connectionStringUrl: ConnectionStringUrl
+): AUTH_TABS {
+  const authMechanismString =
+    connectionStringUrl.searchParams.get('authMechanism');
+
+  const hasPasswordOrUsername =
+    connectionStringUrl.password || connectionStringUrl.username;
+  if (!authMechanismString && hasPasswordOrUsername) {
+    // Default (Username/Password) auth when there is no
+    // `authMechanism` and there's a username or password.
+    return AuthMechanism.MONGODB_DEFAULT;
+  }
+
+  const matchingTab = options.find(({ id }) => id === authMechanismString);
+  if (matchingTab) {
+    return matchingTab.id;
+  }
+
+  switch (authMechanismString) {
+    case AuthMechanism.MONGODB_DEFAULT:
+    case AuthMechanism.MONGODB_SCRAM_SHA1:
+    case AuthMechanism.MONGODB_SCRAM_SHA256:
+    case AuthMechanism.MONGODB_CR:
+      // We bundle SCRAM-SHA-1 and SCRAM-SHA-256 into the Username/Password bucket.
+      return AuthMechanism.MONGODB_DEFAULT;
+    default:
+      return 'AUTH_NONE';
+  }
+}
+
 function AuthenticationTab({
+  errors,
   updateConnectionFormField,
   connectionStringUrl,
 }: {
   errors: ConnectionFormError[];
   connectionStringUrl: ConnectionStringUrl;
   updateConnectionFormField: UpdateConnectionFormField;
-  connectionOptions?: ConnectionOptions;
+  connectionOptions: ConnectionOptions;
 }): React.ReactElement {
-  const selectedAuthMechanism =
-    connectionStringUrl.searchParams.get('authMechanism') ?? '';
+  const selectedAuthTabId =
+    getSelectedAuthTabForConnectionString(connectionStringUrl);
   const selectedAuthTab =
-    options.find(({ id }) => id === selectedAuthMechanism) ?? options[0];
+    options.find(({ id }) => id === selectedAuthTabId) || options[0];
 
   const optionSelected = useCallback(
     (event: ChangeEvent<HTMLInputElement>) => {
       event.preventDefault();
-      updateConnectionFormField({
-        type: 'update-search-param',
-        currentKey: 'authMechanism',
-        value: event.target.value,
+
+      if (event.target.value === 'AUTH_NONE') {
```

Comment:
just a super silly thing, probably this can be done with less code:

``` js
return updateConnectionFormField({
   type: 'update-auth-mechanism',
   authMechanism: event.target.value === 'AUTH_NONE' ? null : event.target.value as AuthMechanism,
});
```